core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -280,6 +280,8 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
 
   private[spark] def ui: Option[SparkUI] = _ui
 
+  def uiWebUrl: Option[String] = _ui.map(_.webUrl)
```

Comment:
I think that's reasonable. I suppose this could be a java.net.URI but no big deal